sdk/identity/azure-identity/tests/test_win_vscode_credential.py

```diff
@@ -0,0 +1,27 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import sys
+import pytest
+if sys.platform.startswith('win'):
+    from azure.identity._credentials.win_vscode_credential import _read_credential
+    import win32cred
+
+@pytest.mark.skipif(not sys.platform.startswith('win'), reason="This test only runs on Windows")
+def test_win_vscode_credential():
+    service_name = u"VS Code Azure"
+    account_name = u"Azure"
+    target = u"{}/{}".format(service_name, account_name)
+    comment = u"comment"
+    token_written = u"test_refresh_token"
+    user_name = u"Azure"
+    credential = {"Type": 0x1,
+                   "TargetName": target,
+                   "UserName": user_name,
+                   "CredentialBlob": token_written,
+                   "Comment": comment,
+                   "Persist": 0x2}
+    win32cred.CredWrite(credential)
+    token_read = _read_credential(service_name, account_name)
```

Comment:
This tests the credential's internals. Can we instead test its public surface?

Reply:
We don't want to expose _read_credential to users. We only expose get_token which is tested by another test.

Reply:
What I mean is, this directly tests behavior of the internal method `_read_credential`. Is it feasible to test that behavior indirectly, through the public API (`get_token`) instead?

Reply:
I use mock get_token in our test so no matter _read_credential returns (as long as it is not none), get_token always returns same thing.

Reply:
And get_token is nothing but redeem the refresh_token to secret. So the key is if _read_credential can get correct refresh_token.